File: 1319-UniqueNumberOfOccurrences/1319-UniqueNumberOfOccurrences.cpp
// Last updated: 3/30/2026, 2:12:34 PM
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> hashmap;
        for(int num : arr) hashmap[num]++;
        vector<int> freq;
        for(auto val : hashmap) freq.push_back(val.second);
        set<int> seen;
        for(int val : freq) seen.insert(val);
        return seen.size()==freq.size();
    }
};

File: 1319-UniqueNumberOfOccurrences/1319-UniqueNumberOfOccurrences.java
// Last updated: 3/30/2026, 2:12:30 PM
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer> freq = new HashMap<>();
        for(int num : arr) freq.put(num,freq.getOrDefault(num,0)+1);
        HashSet<Integer> seen = new HashSet<>();
        for(int num : freq.values()){
            seen.add(num);
        }
        return freq.size()==seen.size();
    }
}